### ISSUE TITLE
[dvsim] Honor BUILD_ROOT for placing scratch directory

### ISSUE THREAD

**Issue by imphil** [lowRISC/opentitan#5031]
BUILD_ROOT is the designated environment variable for build outputs, see `util/build_consts.sh`. In CI environments, we set this variable to a directory outside of the source tree. dvsim currently uses SCRATCH_ROOT instead. 

We should
- Improve dvsim to honor BUILD_ROOT
- Source util/build_consts.sh before running dvsim (e.g. in lint jobs)

**Comment by sriyerg** [comment]
Sure, fine with me. It just won't be in $REPO_TOP though.

**Comment by imphil** [comment]
Yeah, it needs to be a subdirectoy of BUILD_ROOT, ideally starting with `build-`. We originally planned to have $OBJ_DIR used for that purpose, but meson has occupied that directory now for itself, so we need to find another subdirectory for dvsim (or move meson intermediate build artifacts into $OBJ_DIR/meson or something like that).

**Comment by sriyerg** [comment]
Ohh, just remembered why its not as easy: https://github.com/lowRISC/opentitan/pull/5025#issuecomment-769943392
Will think about it though.

**Comment by rswarbrick** [comment]
Let's keep the ability to have BUILD_ROOT / SCRATCH_ROOT inside the repository. It's always a pain to build stuff that doesn't let you do that (because you have to mess around with more directories when checking stuff out - you can't just "git clone foo; cd foo; make-thing.sh").
